### PR TITLE
[Core] Using `VariableData` for sake of consistency instead of `Variable<TDataType>`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This is more consistent with base definitions, where only `VariableData` is considered, additionally removed unnecessary template arguments

**🆕 Changelog**

- [Using `VariableData` for sake of consistency instead of Variable](https://github.com/KratosMultiphysics/Kratos/commit/8ac69ca28e1d5222df6a5683c359e60c747a02d8)
- `pragma once`
- Header clean up
- Remove unused includes

### CODE REVIEW

**Review by rubenzorrilla — DISMISSED** [review]

File: kratos/containers/variables_list_data_value_container.h [hunk 1]
```diff
@@ -4,21 +4,16 @@
...
 
-#if !defined(KRATOS_VARIABLES_LIST_DATA_VALUE_CONTAINER_H_INCLUDED )
-#define KRATOS_VARIABLES_LIST_DATA_VALUE_CONTAINER_H_INCLUDED
+#pragma once
 
 // System includes
-#include <string>
-#include <iostream>
-#include <cstddef>
-#include <cstring>
```
Comment (roigcarlo) [inline kratos/containers/variables_list_data_value_container.h]: Why did you delete the includes? It uses
- `<cstddef>`/ `<cstring>` (line 76)
- `<string>` (line 722) 
- `<iostream>` (line 976)  

By removing it from here you are creating a include dependency on any of the includes below...

**Review by roigcarlo — CHANGES_REQUESTED** [review]
Block for the includes

File: kratos/containers/variables_list_data_value_container.h (hunk 1 above)
Comment (roigcarlo) [reply kratos/containers/variables_list_data_value_container.h]: But `global_variables.h` could probably go away...

Comment (loumalouomega) [reply kratos/containers/variables_list_data_value_container.h]: Reverting

**Comment by loumalouomega** [comment]
> Block for the includes

Done

**Review by roigcarlo — APPROVED** [review]
